src/Core/Joins.cpp

```diff
@@ -0,0 +1,55 @@
+#include <Core/Joins.h>
+
+namespace DB
+{
+
+const char * toString(JoinKind kind)
```

Comment:
Is `magic_enum` intentionally not used?

Reply:
Yes. It had some issues with lifetime of enum entry. We have workaround in ASTSystemQuery.cpp, but here it is simpler to just manually convert to string.